modules/flann/include/opencv2/flann/params.h

```diff
@@ -45,14 +45,18 @@ typedef std::map<cv::String, any> IndexParams;
 
 struct SearchParams : public IndexParams
 {
-    SearchParams(int checks = 32, float eps = 0, bool sorted = true )
+    SearchParams(int checks = 32, float eps = 0, bool sorted = true, bool explore_all_trees = false )
```

Comment:
```
SearchParams(int checks = 32, float eps = 0, bool sorted = true)
{
    init(checks, eps, sorted, false);
}
SearchParams(int checks, float eps, bool sorted, bool explore_all_trees)
{
    init(checks, eps, sorted, explore_all_trees);
}
void init(int checks, float eps, bool sorted, bool explore_all_trees)
{
    ... current implementation ...
}
```

Reply:
I've made the changes. However, if we can get along with two constructors for time being, shouldn't we keep the interface as simple as possible on longer term?

Reply:
Thank you for update!
I believe we can keep one constructor on the master branch (no strict ABI requirement for OpenCV 4.x, only source level API compatibility).
Lets create a separate PR after merging this code to master (merging this PR + related "Merge 3.4" PR).

Reply:
There is another PR for 3.4 where I just do the early stopping when the number of checks is met for a tree, instead of descending all the trees but not taking into account their leaves when the criterium is already met. In this way there is no change in the interface on 3.4. 
However, if it's ok to add a new inteface (the 4 args one) in addition to the current one with 3 args on 3.4, then let's do it. I'm asking that because I know the principle is not to add any new feature in previous versions...
For this current PR on master, the ABI was complaining with the 4 arguments ctor replacing the 3 args one.